File: app/src/main/java/com/example/myapplication/Fragments/AllStoriesFragment.java
package com.example.myapplication.Fragments;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.myapplication.Helpers.EqualSpacingItemDecoration;
import com.example.myapplication.Models.Story;
import com.example.myapplication.R;
import com.example.myapplication.Adapters.AllStoriesAdapter;
import com.example.myapplication.ViewModels.StoryViewModel;
import java.util.List;

public class AllStoriesFragment extends Fragment {

    private StoryViewModel storyViewModel;
    private RecyclerView recyclerView;
    private View view;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.all_stories_fragment, null);

        final TextView textView = view.findViewById(R.id.nostorytext);

        recyclerView = view.findViewById(R.id.allstoriesrecyclerview);
        recyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2));
        recyclerView.setHasFixedSize(true);
        recyclerView.addItemDecoration(new EqualSpacingItemDecoration(16, EqualSpacingItemDecoration.HORIZONTAL));

        try {
            final AllStoriesAdapter allStoriesAdapter = new AllStoriesAdapter(getContext());
            recyclerView.setAdapter(allStoriesAdapter);

            /** Get story view model and show data */
            storyViewModel = ViewModelProviders.of(this).get(StoryViewModel.class);
            storyViewModel.getAllStories().observe(this, new Observer<List<Story>>() {
                @Override
                public void onChanged(List<Story> stories) {
                    allStoriesAdapter.setStories(stories);
                    if(stories.size() == 0) {
                        textView.setVisibility(View.VISIBLE);
                    } else {
                        textView.setVisibility(View.INVISIBLE);
                    }
                }
            });
            /** Get story view model and show data */

        } catch (Exception e) {

        }

        return view;
    }


}

File: app/src/main/java/com/example/myapplication/fragments/AllStoriesFragment.kt
package com.example.myapplication.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.myapplication.adapters.AllStoriesAdapter
import com.example.myapplication.helpers.EqualSpacingItemDecoration
import com.example.myapplication.models.Story
import com.example.myapplication.R

import com.example.myapplication.viewmodels.StoryViewModel
import kotlinx.android.synthetic.main.all_stories_fragment.view.*

class AllStoriesFragment : Fragment() {
    lateinit var storyViewModel: StoryViewModel
    lateinit var recyclerView: RecyclerView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        var view = inflater.inflate(R.layout.all_stories_fragment, null)
        val textView = view.nostorytext
        recyclerView = view.findViewById(R.id.allstoriesrecyclerview)

        recyclerView.layoutManager = GridLayoutManager(context, 2)
        recyclerView.addItemDecoration(EqualSpacingItemDecoration(16, EqualSpacingItemDecoration.HORIZONTAL))
        recyclerView.setHasFixedSize(true)
        val allStoriesAdapter = context?.let { AllStoriesAdapter(it) }
        recyclerView.setAdapter(allStoriesAdapter)

        /** Get story view model and show data  */
        storyViewModel = ViewModelProvider(this).get(StoryViewModel::class.java)
        storyViewModel.allStories!!.observe(viewLifecycleOwner, Observer<List<Story?>?> { stories ->
            if (allStoriesAdapter != null) {
                allStoriesAdapter.setStories(stories as MutableList<Story>)
                if (stories.size == 0) {
                    textView.visibility = View.VISIBLE
                } else {
                    textView.visibility = View.INVISIBLE
                }
            }
        })
        /** Get story view model and show data  */

        return view
    }
}